### ISSUE TITLE
Making TRANSACTION_PENDING_LIMIT, AGGREGATE_PENDING_LIMIT configurable

### ISSUE THREAD

**Issue by sjmittal** [glowroot/glowroot#686]
In the code we have:
```
    // back pressure on transaction collection
    private static final int TRANSACTION_PENDING_LIMIT = 1000;
    // back pressure on writing captured data to disk/network
    private static final int AGGREGATE_PENDING_LIMIT = 5;
```
Also the default value for configurable parameter:
`glowroot.internal.rollup.0.intervalMillis` is 60000 ms.

In case we decide to configure this parameter to a lower value say betwwn 1 - 100 ms I think the above two properties defined in `TransactionProcessor` would also have to be changed.
 Based on my understanding if we reduce `glowroot.internal.rollup.0.intervalMillis` we would also have to reduce `TRANSACTION_PENDING_LIMIT` and increase `AGGREGATE_PENDING_LIMIT`.

Let me know if it make sense to have these configurable?

**Comment by Never-lab** [comment]
Wishlist: make `TRANSACTION_PENDING_LIMIT` / `AGGREGATE_PENDING_LIMIT` (or equivalents) **configurable**. Related backlog throttling (#565 / #585). Enhancement.

**Comment by Never-lab** [comment]
Yes — makes sense to configure them the same way as `glowroot.internal.rollup.0.intervalMillis` (JVM system properties), not as UI Advanced config.

**Where:** `TransactionProcessor` — today hardcoded `1000` / `5`.

**Proposed knobs (startup only; queue size is fixed at agent start):**

| Property | Default | Role |
|----------|---------|------|
| `glowroot.internal.transactionPendingLimit` | `1000` | Max completed transactions waiting to be aggregated. Drop + warn when full. |
| `glowroot.internal.aggregatePendingLimit` | `5` | Max finished interval collectors waiting to flush to disk/network. Drop aggregate + warn when full. |

**Example** (with a short rollup interval):

```
-Dglowroot.internal.rollup.0.intervalMillis=100
-Dglowroot.internal.transactionPendingLimit=200
-Dglowroot.internal.aggregatePendingLimit=20
```

Shorter intervals → more interval collectors in flight → raise `aggregatePendingLimit` if you see “not storing an aggregate because of an excessive backlog”. If the processing thread can’t keep up with completions, raise or lower `transactionPendingLimit` depending on whether you prefer buffering vs dropping.

Related: #565 / #585 (broader throttling).

**Comment by Never-lab** [comment]
Nice idea — thanks for filing it.

Closing as **obsolete**: open for years with no concrete design/PR path against current 0.14.x, and we are not planning to implement it as originally scoped. If this is still needed, please open a fresh issue with a short proposal against a current Glowroot build.

**Brief analysis:** Making pending limits configurable is reasonable; still no agreed defaults/UI surface after years—file anew with proposed knobs if priority.
